src/bindings/python/flux/rpc.py

```diff
@@ -10,16 +10,39 @@
 
 import json
 
+from flux.util import check_future_error
 from flux.wrapper import Wrapper
 from flux.future import Future
-from flux.core.inner import ffi, raw
+from flux.core.inner import ffi, lib, raw
 import flux.constants
 from flux.util import encode_payload, encode_topic
 
 
 class RPC(Future):
     """An RPC state object"""
 
+    class RPCInnerWrapper(Wrapper):
+        def __init__(
+            self,
+            handle=None,
+            match=ffi.typeof("flux_future_t *"),
+            filter_match=True,
+            prefixes=None,
+            destructor=raw.flux_future_destroy,
+        ):
+            # avoid using a static list as a default argument
+            # pylint error 'dangerous-default-value'
+            if prefixes is None:
+                prefixes = ["flux_rpc_", "flux_future_"]
+
+            super().__init__(
+                ffi, lib, handle, match, filter_match, prefixes, destructor
+            )
+
+        def check_wrap(self, fun, name):
+            func = super().check_wrap(fun, name)
+            return check_future_error(func)
+
```

Comment:
I thought your previous `pimpl_t` solution was quite clever.  It nicely avoided duplication.  What was the issue that forced this solution?  I'm totally fine with it, but I'm curious as to the underlying cause.

Reply:
In that version RPCInnerWrapper().get could, if Future were initialized first, find `flux_future_get` instead of `flux_rpc_get`, or anything similar where there's a conflict on the tail.  This happened because of inheritence.  I had thought when I did the first one that we didn't have this kind of conflict, but apparently we do, so I had to make them disjoint.